### PR TITLE
twtxt: init at 0.1.0

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Add twtxt

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 105378` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>twtxt</li>
  </ul>
</details>

File: pkgs/applications/networking/twtxt/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "15jhfnhpk34nmad04f7xz1w041dba8cn17hq46p9n5sarjgkjiiw";
+  };
+
+  vendorSha256 = "1lnf8wd2rv9d292rp8jndfdg0rjs6gfw0yg49l9spw4yzifnd7f7";
+
+  # Seems to have been put there during build.
+  postInstall = ''
+    rm $out/bin/tests
```
Comment (kalbasit) [inline pkgs/applications/networking/twtxt/default.nix:18]: Another way to do this (preferably) is to specify the subPackages to restrict what it's building

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/twtxt/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/twtxt/default.nix:18]:
```
subPackages = [ "binary" ];```

File: pkgs/applications/networking/twtxt/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Self-hosted, Twitter-like decentralised microblogging platform";
+    homepage = "https://github.com/jointwt/twtxt";
+    license = licenses.mit;
+    maintainers = with maintainers; [ siraben ];
+    platforms = platforms.unix;
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/networking/twtxt/default.nix:26]: Please remove this.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>twtxt</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105378` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>twtxt</li>
  </ul>
</details>

**Review by kalbasit — APPROVED** [review]

**Pull request merged**